### ISSUE TITLE
Random forests hyper parameters

### ISSUE THREAD

**Issue by emilemathieu** [cianeastwood/qedr#2]
Hi !
Thanks for the great repo :)

Could you please specify how you chose (optimised which criterion) the hyper parameters `all_best_depths` of the random forests ?

Cheers,
Emile

**Comment by cianeastwood** [comment]
Hi Emile,

Tree depth was chosen to achieve the lowest prediction error on the validation set. (See the last sentence on page 5 of our paper [section 4.3]).

Best,
Cian

**Comment by emilemathieu** [comment]
Hi Cian,

Thanks for your answer! Therefore you train the regressor on the training dataset (same part of the dataset the model has been trained on), and optimise the regressor hyperparameters on the validation set if I understand correctly.

Also, isn't a bit annoying that the beta-VAE achieves better disentanglement than the VAE according to the random forest regressor but worse according to the lasso regressor ?

Cheers,
Emile

**Comment by emilemathieu** [comment]
Also, related to the training of the regressors, to get `z_hat` do you take the mean of the encoder `qz_x` or do you sample from that posterior distribution ?

Cheers,
Emile

**Comment by cianeastwood** [comment]
Hi Emile,

Sincere apologies -- I must have missed this!

I'm not sure if it is still relevant, but yes we do use the training and validation sets exactly as you suggested.

As for the change in relative disentanglement scores across regressors, it is important to note that these scores depend on (i) how "easy" it is to extract the information about z in c, and (ii) whether or not the chosen regressor has the capacity to extract all of this information. This invokes some interesting thoughts about these metrics, some of which I am currently exploring.

I am not sure what `z_hat` refers to here. The image code (**c**)/encoding/embedding? Let me know if it is still relevant and I can try to clarify.

Best,
Cian
